backend/src/routes.rs

```diff
@@ -0,0 +1,154 @@
+use actix_web::{
+    delete, get, post, put,
+    web::{self},
+    HttpResponse, Scope,
+};
+use subxt::ext::sp_core::H256;
+use uuid::Uuid;
+
+use crate::{
+    database::{
+        dto::{AttestationRequest, Credential, Query, UpdateAttestation},
+        querys::{
+            approve_attestation_request, can_approve_attestation, can_revoke_attestation,
+            delete_attestation_request, get_attestation_request_by_id, get_attestation_requests,
+            get_attestations_count, insert_attestation_request, revoke_attestation_request,
+            update_attestation_request,
+        },
+    },
+    error::AppError,
+    AppState,
+};
+
+#[get("/{attestation_request_id}")]
+async fn get_attestation(
+    path: web::Path<Uuid>,
+    state: web::Data<AppState>,
+) -> Result<HttpResponse, AppError> {
+    let attestation_id = path.into_inner();
+
+    let attesttation = get_attestation_request_by_id(&attestation_id, &state.db_executor).await?;
+
+    Ok(HttpResponse::Ok().json(serde_json::json!(attesttation)))
+}
+
+#[get("")]
+async fn get_attestations(
+    state: web::Data<AppState>,
+    pagination_query: web::Query<Query>,
+) -> Result<HttpResponse, AppError> {
+    let pagination = pagination_query.into_inner().into();
+
+    let content_range = get_attestations_count(&state.db_executor).await;
+
+    let attestation_requests = get_attestation_requests(pagination, &state.db_executor).await?;
+
+    let response = serde_json::json!(attestation_requests);
+
+    Ok(HttpResponse::Ok()
+        .insert_header(("Content-Range", content_range))
+        .json(response))
+}
+
+#[delete("/{attestation_request_id}")]
+async fn delete_attestation(
+    path: web::Path<Uuid>,
+    state: web::Data<AppState>,
+) -> Result<HttpResponse, AppError> {
+    let attestation_id = path.into_inner();
+    delete_attestation_request(&attestation_id, &state.db_executor).await?;
+    log::info!("Attestation with id {:?} is deleted", attestation_id);
+    Ok(HttpResponse::Ok().json("ok"))
+}
+
+#[post("")]
+async fn post_attestation(
+    body: web::Json<AttestationRequest>,
+    state: web::Data<AppState>,
+) -> Result<HttpResponse, AppError> {
+    let attestation_request = body.into_inner();
+    let credential: Credential = serde_json::from_value(attestation_request.clone().credential)?;
+    let attestation =
+        insert_attestation_request(&attestation_request, &credential, &state.db_executor).await?;
+    log::info!(" New attestation with id {:?} is created", attestation.id);
+    Ok(HttpResponse::Ok().json(attestation))
+}
+
+#[put("/{attestation_request_id}/approve")]
+async fn approve_attestation(
+    path: web::Path<Uuid>,
+    state: web::Data<AppState>,
+) -> Result<HttpResponse, AppError> {
+    let attestation_id = path.into_inner();
+    let attestation = can_approve_attestation(&attestation_id, &state.db_executor).await?;
+    let credential = serde_json::from_value::<Credential>(attestation.credential)?;
```

Comment:
You sometimes do `let credential: Credential = ...` and sometimes the `::<Credential>`. I would recommend to stick to one of the two options and I prefer the first one.

```suggestion
    let credential: Credential = serde_json::from_value(attestation.credential)?;
```

Reply:
Made it consistent: [a0d7b04](https://github.com/KILTprotocol/attester/pull/1/commits/a0d7b042cf735da6244f8c390298eee8d4195dae)